nifi-framework-api/src/main/java/org/apache/nifi/authorization/AbstractPolicyBasedAuthorizer.java

```diff
@@ -149,6 +152,10 @@ public final synchronized Group addGroup(Group group) throws AuthorizationAccess
      */
     public abstract Group getGroup(String identifier) throws AuthorizationAccessException;
 
+    public abstract void purgePoliciesUsersAndGroups();
+
+    public abstract void backupPoliciesUsersAndGroups();
```

Comment:
If backup and purge are not part of the formal API in the `ConfigurableUserGroupProvider` and `ConfigurableAccessPolicyProvider` does it need to be a public method on the `AbstractPolicyBasedAuthorizer`?

Reply:
Good call, they should be `protected`